src/dos/drive_local.cpp

```diff
@@ -626,17 +646,18 @@ bool localFile::Seek(uint32_t *pos, uint32_t type)
 	fpos_t temppos;
 	fgetpos(fhandle,&temppos);
 	Bit32u * fake_pos=(Bit32u*)&temppos;
-	*pos=*fake_pos;
+	*pos_addr = *fake_pos;
 #endif
-	if (ftell_and_check())
-		*pos = static_cast<uint32_t>(stream_pos);
-
-	// If ftell fails then stream_pos becomes -1 however this function's
-	// 'pos' argument can't handle -1 because it's unsigned. So we
-	// explicitly underflow it to its maximum to acknowledge what's
-	// actually happening:
-	else
-		*pos = UINT32_MAX;
+	static_cast<void>(ftell_and_check());
```

Comment:
I believe this `ftell_and_check` call is no longer needed?

Reply:
We need to `ftell` to update the `stream_pos` member, which we then write back into the `*pos_addr` below:

`*reinterpret_cast<int32_t *>(pos_addr) = static_cast<int32_t>(stream_pos);`